persistence-cassandra-3.11/src/main/java/io/stargate/db/cassandra/impl/Conversion.java

```diff
@@ -187,33 +202,38 @@ public static ByteBuffer toPagingState(PagingPosition pos, Parameters parameters
     }
 
     Object[] pkValues =
-        cfm.partitionKeyColumns().stream()
-            .map(
-                c -> {
-                  ByteBuffer value = pos.currentRowValuesByColumnName().get(c.name.toCQLString());
-
-                  if (value == null) {
-                    throw new IllegalArgumentException(
-                        String.format(
-                            "Partition key value is not present in current row (table: %s, column: %s)",
-                            table, c.name.toCQLString()));
-                  }
-
-                  return value;
-                })
-            .toArray();
-    Clustering clustering = cfm.getKeyValidatorAsClusteringComparator().make(pkValues);
-
-    if (pos.resumeFrom() != ResumeMode.NEXT_PARTITION) {
-      throw new UnsupportedOperationException("Unsupported paging mode: " + pos.resumeFrom());
+        cfm.partitionKeyColumns().stream().map(c -> getKeyValue(pos, cfm, c)).toArray();
+    Clustering partitionKey = cfm.getKeyValidatorAsClusteringComparator().make(pkValues);
+    ByteBuffer serializedKey = CFMetaData.serializePartitionKey(partitionKey);
+
+    org.apache.cassandra.transport.ProtocolVersion protocolVersion =
+        toInternal(parameters.protocolVersion());
+
+    RowMark rowMark;
+    switch (pos.resumeFrom()) {
+      case NEXT_PARTITION:
+        rowMark = null;
+        break;
+
+      case NEXT_ROW:
+        ByteBuffer[] ccValues =
+            cfm.clusteringColumns().stream()
+                .map(c -> getKeyValue(pos, cfm, c))
+                .toArray(ByteBuffer[]::new);
+
+        Clustering rowClustering = Clustering.make(ccValues);
+        BTreeRow row = BTreeRow.noCellLiveRow(rowClustering, LivenessInfo.create(0, 0));
```

Comment:
what's `LivenessInfo.create(0, 0)`? It seems that it is on the hot path, maybe extract to constant?

Reply:
It's a required parameter for `Row` but it's not used for constructing the `PagingState`. Will extract.

Reply:
extracted to a `private` constant.